### PR TITLE
Switch to YAXArrays when runnin scenarios

### PR DESCRIPTION
**Pull request by Zapiano**
As in title. This one should be merged AFTER [PR 689](https://github.com/open-AIMS/ADRIA.jl/pull/689).

### CODE REVIEW

**Comment by ConnectedSystems** [comment]
> As in title. This one should be merged AFTER [PR 689](https://github.com/open-AIMS/ADRIA.jl/pull/689).

FYI, you don't need to manually create the links, just tag the PR ID after a hash, e.g., `#689` -> #689

File: src/scenario.jl [hunk 1]
```diff
@@ -89,7 +89,7 @@ function run_scenarios(
     setup()
 
     # Sort RCPs so the dataframe order match the output filepath
-    RCP = Base.sort(RCP)
+    RCP = sort(RCP)
```
Comment (ConnectedSystems) [inline src/scenario.jl:92]: Was this intentional?

File: src/scenario.jl [hunk 2]
```diff
@@ -329,19 +335,19 @@ function run_model(domain::Domain, param_set::DataFrameRow)::NamedTuple
...
-function run_model(domain::Domain, param_set::NamedDimsArray)::NamedTuple
+function run_model(domain::Domain, param_set::YAXArray)::NamedTuple
     p = domain.coral_growth.ode_p
     corals = to_coral_spec(param_set)
     cache = setup_cache(domain)
 
     # Set random seed using intervention values
     # TODO: More robust way of getting intervention/criteria values
-    rnd_seed_val::Int64 = floor(Int64, sum(param_set(!=("RCP"))))  # select everything except RCP
+    rnd_seed_val::Int64 = floor(Int64, sum(param_set[Where(x -> x != "RCP")]))  # select everything except RCP
```
Comment (ConnectedSystems) [inline src/scenario.jl:345]: Not a big deal, but does this work?

```julia
not_RCP = param_set.factors .!= "RCP"
param_set[factors=not_RCP]
```

File: src/scenario.jl [hunk 3]
```diff
@@ -329,19 +335,19 @@ function run_model(domain::Domain, param_set::DataFrameRow)::NamedTuple
...
     Random.seed!(rnd_seed_val)
 
-    dhw_idx::Int64 = Int64(param_set("dhw_scenario"))
-    wave_idx::Int64 = Int64(param_set("wave_scenario"))
-    cyclone_mortality_idx::Int64 = Int64(param_set("cyclone_mortality_scenario"))
+    dhw_idx::Int64 = Int64(param_set[At("dhw_scenario")])
```
Comment (ConnectedSystems) [inline src/scenario.jl:348]: With these `param_set[At(...)]`, I think `param_set.factor_name` should work and save a few keystrokes.

File: src/scenario.jl (hunk 2 above)
Comment (ConnectedSystems) [reply src/scenario.jl:345]: The reason is I prefer to avoid lambdas (comparatively slow).

File: src/scenario.jl (hunk 1 above)
Comment (Zapiano) [reply src/scenario.jl:92]: Yes. This is how it is on the main branch currently. I had changed to `Base.sort(RCP)` in #689 but that is not necessary anymore.

File: src/scenario.jl (hunk 3 above)
Comment (Zapiano) [reply src/scenario.jl:348]: I think that doesn't work. What I *can* do, is `param_set.factors` , becausa `:factors` is the name of an axis - but that doesn't return what we want here. The names of the factors are *labels* of that axis. I can think of two ways of selecting one single axis element in this case:

```
param_set[param_set.factors .== "dhw_scenario"].data[1]
param_set[At("dhw_scenario")]
```

image

The former is useful when we have more axes (I'm not sure if/how the later works with more axes).

File: src/decision/dMCDA.jl [hunk 4]
```diff
@@ -3,9 +3,11 @@ module decision
 
 using StatsBase
 using Distances, DataFrames
+using DimensionalData
```
Comment (ConnectedSystems) [inline src/decision/dMCDA.jl:6]: Is this import necessary?

If it's just for the `At` method, I prefer it's imported directly:

`import YAXArrays.DD: At`

**Review by ConnectedSystems — APPROVED** [review]
Thanks

File: src/decision/dMCDA.jl (hunk 4 above)
Comment (Zapiano) [reply src/decision/dMCDA.jl]: Apparently it doesn't make any difference (I've removed it and everything worked fine).

**Pull request merged**